geoapps/processing/peak_finder.py

```diff
@@ -29,206 +31,135 @@
-    SelectMultiple,
     ToggleButton,
     ToggleButtons,
     VBox,
+    Widget,
     interactive_output,
 )
-from scipy.spatial import cKDTree
+from ipywidgets.widgets.widget_selection import TraitError
 
+from geoapps.base import BaseApplication
 from geoapps.selection import LineOptions, ObjectDataSelection
 from geoapps.utils import geophysical_systems
-from geoapps.utils.utils import (
-    LineDataDerivatives,
-    colors,
-    find_value,
-    hex_to_rgb,
-    rotate_azimuth_dip,
-    running_mean,
-)
+from geoapps.utils.formatters import string_name
+from geoapps.utils.utils import LineDataDerivatives, hex_to_rgb, running_mean
+
+from ..io.PeakFinder import PeakFinderParams
+from ..io.PeakFinder.constants import default_ui_json
+
+_default_channel_groups = {
+    "early": {"label": ["early"], "color": "#0000FF", "channels": []},
+    "middle": {"label": ["middle"], "color": "#FFFF00", "channels": []},
+    "late": {"label": ["late"], "color": "#FF0000", "channels": []},
+    "early + middle": {
+        "label": ["early", "middle"],
+        "color": "#00FFFF",
+        "channels": [],
+    },
+    "early + middle + late": {
+        "label": ["early", "middle", "late"],
+        "color": "#008000",
+        "channels": [],
+    },
+    "middle + late": {
+        "label": ["middle", "late"],
+        "color": "#FFA500",
+        "channels": [],
+    },
+}
 
 
 class PeakFinder(ObjectDataSelection):
     """
     Application for the picking of targets along Time-domain EM profiles
     """
 
-    defaults = {
-        "h5file": "../../assets/FlinFlon.geoh5",
-        "objects": "{bb208abb-dc1f-4820-9ea9-b8883e5ff2c6}",
-        "data": ["Observed"],
-        "model": {
-            "objects": "{2e814779-c35f-4da0-ad6a-39a6912361f9}",
-            "data": "Iteration_7_model",
-        },
-        "lines": {
-            "objects": "{bb208abb-dc1f-4820-9ea9-b8883e5ff2c6}",
-            "data": "Line",
-            "lines": 6073400.0,
-        },
-        "boreholes": {"objects": "geochem", "data": "Al2O3"},
-        "doi": {"data": "Z"},
-        "doi_percent": 60,
-        "doi_revert": True,
-        "center": 4050,
-        "width": 1000,
-        "smoothing": 6,
-        "tem_checkbox": True,
-        "markers": True,
-        "slice_width": 25,
-        "x_label": "Distance",
-        "ga_group_name": "PeakFinder",
-    }
-
-    _add_groups = True
+    _param_class = PeakFinderParams
+    _add_groups = "only"
     _object_types = (Curve,)
+    _group_auto = None
+    decay_figure = None
+    marker = {"left": "<", "right": ">"}
 
-    def __init__(self, **kwargs):
-
-        self.defaults = self.update_defaults(**kwargs)
-
-        try:
-            self.client = get_client()
-        except ValueError:
-            self.client = Client()
+    def __init__(self, ui_json=None, **kwargs):
+        if ui_json is not None and path.exists(ui_json):
+            self.params = self._param_class.from_path(ui_json)
+        else:
```

Comment:
Can the else contents be replaced from a self._param_class.from_dict(default_dict) call?

Reply:
Oh, I see, you're allowing the __init__ to update the default_dict and THEN you call the .from_dict constructor.  Good thinking with the try/except block there.

Reply:
Maybe these details could be included in the from_dict definition in io/params.py so that they can be used elsewhere?

Reply:
So do a `from_dict` that accepts kwargs?

Reply:
Done